openedx/core/djangoapps/content/course_overviews/models.py

```diff
@@ -265,6 +266,39 @@ def get_from_id(cls, course_id):
             course_overview = None
         return course_overview or cls.load_from_module_store(course_id)
 
+    def get_course_status(self, cert_status):
```

Comment:
Is this method `get_course_status` semantically specific to certificates?  Currently, it does seem that way.  If so, I'd recommend moving it to a certificate-specific library (along with 'may_certify').  Or, at the least, renaming it to something like 'get_certificate_status`.

My inclination is to keep only the course data model in the course_overviews.models module.  All other business logic (in this case, specific to certificates/status), should be at a higher layer.